src/ops.rs

```diff
@@ -77,9 +77,11 @@ pub fn popa(vm: &mut VM, pipeline: &Pipeline, _hv: &mut dyn Hypervisor) -> Resul
 ///  The logic function for the 'enter' opcode
 pub fn enter(vm: &mut VM, pipeline: &Pipeline, _hv: &mut dyn Hypervisor) -> Result<(), VMError>{
     let locals = vm.get_arg(pipeline.args[0].location)?.u16_exact()?;
-    let mut nesting = vm.get_arg(pipeline.args[1].location)?.u8_exact()?;
+    vm.eip += 2;
```

Comment:
Because EIP is changed directly, this instruction should be marked as unpredictable for pipeline purposes. I also don't really understand why EIP is changed here

Reply:
ah shoot I accidentally kept this in here while trying to copy the logic from the cpp version. I'll fix it up.